### PR TITLE
feat: support webauthn login

### PR DESCRIPTION
**Pull request by itsHenry35**

### CODE REVIEW

**Comment by itsHenry35** [comment]
Known issue: /getcredentials won't get the latest webauthn credentials, unless the alist server is restarted
I don't know how to solve this, so I'm seeking help from the community

**Comment by itsHenry35** [comment]
I did manual merge on both frontend and backend, therefore, I'm not really sure if I have missed any changes

**Comment by itsHenry35** [comment]
available binary with frontend: 
[alist_2.zip](https://github.com/alist-org/alist/files/12272642/alist_2.zip)

**Review by xhofe — APPROVED** [review]

**Comment by itsHenry35** [comment]
@xhofe when will it be merged after you approve it?

File: internal/model/user.go [hunk 1]
```diff
@@ -41,6 +46,7 @@ type User struct {
 	Permission int32  `json:"permission"`
 	OtpSecret  string `json:"-"`
 	SsoID      string `json:"sso_id"` // unique by sso platform
+	Authn      string `gorm:"size:4294967295" json:"-"`
```
Comment (xhofe) [inline internal/model/user.go:49]: Why does it need to be so long?

Comment (itsHenry35) [reply internal/model/user.go:49]: > Why does it need to be so long?

even one credential takes around 1000 characters, to be honest, now I also think it's a bit too long, how about let's shorten it to 65535(MySQL Largest)?

Comment (xhofe) [reply internal/model/user.go]: I think you can use `text` type directly like https://github.com/alist-org/alist/blob/0b675d6c024cbaec31807113b5a65f551a289d80/internal/model/storage.go#L12

**Comment by anwen-anyi** [comment]
上次Google无法链接和Apple超时的有缓解吗？我再试试看

File: internal/model/user.go (hunk 1 above)
Comment (itsHenry35) [reply internal/model/user.go]: > gorm:"type:text"

got it, i tested it and type text works fine

**Review by xhofe — APPROVED** [review]

**Pull request merged**